lib/slim_lint/runner.rb

```diff
@@ -19,9 +19,14 @@ def run(options = {})
 
       linter_selector = SlimLint::LinterSelector.new(config, options)
 
-      lints = files.map do |file|
-        collect_lints(file, linter_selector, config)
-      end.flatten
+      lints =
+        if options[:stdin_file_path].nil?
+          files.map do |file|
+            collect_lints(File.read(file), file, linter_selector, config)
+          end.flatten
+        else
+          collect_lints($stdin.read, options[:stdin_file_path], linter_selector, config)
+        end
 
       SlimLint::Report.new(lints, files)
```

Comment:
It seems like we would need to ensure `files` is correctly updated to use the provided `stdin_file_path`, which is missing here. Have we tested this? What do the reports show?

Reply:
Could you clarify how exactly should it be updated? Currently with stdin `files = extract_applicable_files(config, options)` returns an empty array.

This is how it works:
```
$ cat t.slim
divdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdivdiv
  = 1
$ cat  t.slim | bundle exec ./bin/slim-lint --stdin-file-path index.slim
warning: parser/current is loading parser/ruby27, which recognizes
warning: 2.7.3-compliant syntax, but you are running 2.7.2.
warning: please see https://github.com/whitequark/parser#compatibility-with-ruby-mri.
index.slim:1 [W] LineLength: Line is too long. [105/80]
```

or in short
```
index.slim:1 [W] LineLength: Line is too long. [105/80]
```

Reply:
Right, you don't want to pass an empty `files` array to `SlimLint::Report.new(lints, files)`. You would want to pass a single item array containing `[options[:stdin_file_path]]`, i.e. `files = [options[:stdin_file_path]]`.

See the definition for the `Report` class. It's expecting the list of files that were linted:
https://github.com/sds/slim-lint/blob/f9c2c2143c35d21c9a65cce82e5c8b8c47c7ee6b/lib/slim_lint/report.rb#L9-L10

Reply:
Thanks! I changed that `files` variable as you asked. For your information, it didn't change the output, everything is still same.

Reply:
Ah, that's probably because the default reporter doesn't use `files`. But it _is_ used by other reporters.